### PR TITLE
Add Feature flag to whitelist users from recaptcha security check (PROJQUAY-3697)

### PR DESCRIPTION
**Pull request by harishsurf**
Add Feature flag to whitelist users from recaptcha security check(PROJQUAY-3697)

Signed-off-by: harishsurf <hgovinda@redhat.com>

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -592,6 +592,9 @@ def create_transaction(db):
     RECAPTCHA_SITE_KEY: Optional[str] = None
     RECAPTCHA_SECRET_KEY: Optional[str] = None
 
+    # List of users allowed to pass through recaptcha security check to enable org/user creation via API
+    RECAPTCHA_WHITELIST_USERS: List[str] = []
```
Comment (kleesc) [inline config.py:596]: RECAPTCHA_WHITELISTED_NAMESPACES

File: endpoints/api/user.py [hunk 2]
```diff
@@ -480,6 +480,7 @@ def put(self):
 
     @show_if(features.USER_CREATION)
     @show_if(features.DIRECT_LOGIN)
+    @require_user_admin
```
Comment (kleesc) [inline endpoints/api/user.py:483]: Why is this added?

**Review by kleesc — COMMENTED** [review]
Otherwise lgtm

File: endpoints/api/user.py (hunk 2 above)
Comment (harishsurf) [reply endpoints/api/user.py:483]: the `get_authenticated_user()` was returning a `NoneType` exception without it

File: config.py (hunk 1 above)
Comment (syed) [reply config.py:596]: should it be `NAMESPACES` instead of `USERS` ?

File: endpoints/api/user.py (hunk 2 above)
Comment (kleesc) [reply endpoints/api/user.py:483]: Which is expected if this is called by an anonymous user.

**Review by kleesc — CHANGES_REQUESTED** [review]

File: endpoints/api/user.py (hunk 2 above)
Comment (kleesc) [reply endpoints/api/user.py:483]: Say when a new user is registering for the first time.

File: test/test_api_usage.py [hunk 3]
```diff
@@ -814,6 +814,11 @@ def captcha_endpoint(url, request):
                 details["recaptcha_response"] = "somecode"
                 self.postResponse(User, data=details, expected_code=200)
 
+    def test_recaptcha_whitelist_users(self):
+        with (self.toggleFeature("RECAPTCHA", True)):
+            app.config["RECAPTCHA_WHITELIST_USERS"] = ["bobby"]
```
Comment (kleesc) [inline test/test_api_usage.py:819]: Can you reference "bobby" as `NEW_USER_DETAILS["username"]` instead?

**Review by kleesc — CHANGES_REQUESTED** [review]

File: config.py (hunk 1 above)
Comment (harishsurf) [reply config.py:596]: wouldn't `Namespaces` include orgs? However this list would only take in `users` as input?

Comment (kleesc) [reply config.py:596]: Right, RECAPTCHA_WHITELISTED_USERS should be fine then.

File: endpoints/api/user.py (hunk 2 above)
Comment (harishsurf) [reply endpoints/api/user.py]: ah, yes, I've updated it. PTAL

File: endpoints/api/user.py [hunk 4]
```diff
@@ -516,13 +516,16 @@ def post(self):
 
         # If recaptcha is enabled, then verify the user is a human.
         if features.RECAPTCHA:
-            recaptcha_response = user_data.get("recaptcha_response", "")
-            result = recaptcha2.verify(
-                app.config["RECAPTCHA_SECRET_KEY"], recaptcha_response, get_request_ip()
-            )
+            user = get_authenticated_user()
+            # check if the user is whitelisted to bypass recaptcha security check
+            if user is None or (user is not None and user.username not in features.RECAPTCHA_WHITELISTED_USERS):
```
Comment (kleesc) [inline endpoints/api/user.py:521]: You don't need `user is not None` since it's a disjunction. Just `user is None or user.username not in features.RECAPTCHA_WHITELISTED_USERS`

**Review by kleesc — CHANGES_REQUESTED** [review]

**Comment by kleesc** [comment]
LGTM otherwise

**Review by kleesc — APPROVED** [review]
LGTM
